cdap-client-tests/src/test/java/co/cask/cdap/client/RouteConfigClientTestRun.java

```diff
@@ -0,0 +1,181 @@
+    addAppReturnsArgsArtifact();
+  }
+
+  @Test
+  public void test() throws Exception {
+    // Deploy AppReturnsArgs of version v1 and start its service
+    String version1 = "v1";
+    ApplicationId app1 = new ApplicationId(NamespaceId.DEFAULT.getNamespace(), AppReturnsArgs.NAME, version1);
+    deployApp(app1);
+    Map<String, String> propMap1 = startServiceForApp(app1);
+
+    // RouteConfig is empty by default
+    Assert.assertTrue(routeConfigClient.getRouteConfig(NamespaceId.DEFAULT,
+                                                       AppReturnsArgs.NAME, AppReturnsArgs.SERVICE).isEmpty());
+
+    // Calls to the non-versioned service endpoint are routed to the only running service of version v1
+    assertArgsEqual(propMap1);
+
+    // Deploy AppReturnsArgs of version v2 and start its service
+    String version2 = "v2";
+    ApplicationId app2 = new ApplicationId(NamespaceId.DEFAULT.getNamespace(), AppReturnsArgs.NAME, version2);
+    deployApp(app2);
+    Map<String, String> propMap2 = startServiceForApp(app2);
+
+    // Storing and getting RouteConfig with total percentage equal to 100 will succeed
+    storeAndGetValidRouteConfig(ImmutableMap.of(version1, 10, version2, 90));
+    storeAndGetValidRouteConfig(ImmutableMap.of(version1, 20, version2, 80));
+    storeAndGetValidRouteConfig(ImmutableMap.of(version1, 60, version2, 40));
+    
+    // Route all calls to the non-versioned service endpoint are routed to the service of v2
+    storeAndGetValidRouteConfig(ImmutableMap.of(version1, 0, version2, 100));
+    assertArgsEqual(propMap2);
+
+    // Setting RouteConfig with total percentage not equal to 100 will fail
+    storeInvalidRouteConfig(ImmutableMap.of(version1, 100, version2, 1), "Percentage");
+    // Setting RouteConfig with total percentage not equal to 100 will fail
+    storeInvalidRouteConfig(ImmutableMap.of(version1, 98, version2, 1), "Percentage");
+    // Setting RouteConfig with non-existing version will fail
+    storeInvalidRouteConfig(ImmutableMap.of(version1, 100, "NONEXISTING", 0), "NONEXISTING");
+
+    // Delete RouteConfig
+    routeConfigClient.deleteRouteConfig(NamespaceId.DEFAULT, AppReturnsArgs.NAME, AppReturnsArgs.SERVICE);
+    // RouteConfig is empty after deletion
+    Assert.assertTrue(routeConfigClient.getRouteConfig(NamespaceId.DEFAULT, AppReturnsArgs.NAME,
+                                                       AppReturnsArgs.SERVICE).isEmpty());
+
+    // Stop all services
+    programClient.stopAll(Id.Namespace.DEFAULT);
+    // Delete all apps
+    appClient.deleteAll(Id.Namespace.DEFAULT);
+  }
+
+  private void deployApp(ApplicationId app) throws Exception {
+    // Deploy app of given version
+    AppRequest createRequest = new AppRequest<>(
+      new ArtifactSummary(artifactId.getArtifact(), artifactId.getVersion()));
+    appClient.deploy(app, createRequest);
+  }
+
+  private Map<String, String> startServiceForApp(ApplicationId app) throws Exception {
```

Comment:
just ``startService``. 

Also, this method still returns runtimeArgs? I thought we discussed that it should take the service and the runtime args, and then start it?